lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -727,7 +732,11 @@ def rank_ecc(
         for rawfc, calfc in zip(
                 raw_forecast_members.slices_over("time"),
                 post_processed_forecast_percentiles.slices_over("time")):
-            random_data = np.random.random(rawfc.data.shape)
+            if fixed_random_seed is not None:
```

Comment:
I don't think you need this check. According to https://docs.scipy.org/doc/numpy/reference/generated/numpy.random.RandomState.html#numpy.random.RandomState
you can simply pass the None through, since it's the default behaviour anyway.